packages/next-tinacms-dos/src/handlers.ts

```diff
@@ -115,19 +123,46 @@ async function uploadMedia(
       id: prefix + filename,
       filename,
       directory: prefix,
-      previewSrc: cdnUrl + prefix + filename,
-      src: cdnUrl + prefix + filename,
+      previewSrc:
+        cdnUrl +
+        (mediaRoot
+          ? path.join(mediaRoot, prefix + filename)
+          : prefix + filename),
+      src:
+        cdnUrl +
+        (mediaRoot
+          ? path.join(mediaRoot, prefix + filename)
+          : prefix + filename),
     })
   } catch (e) {
     res.status(500).send(findErrorMessage(e))
   }
 }
 
+function stripMediaRoot(mediaRoot: string, key: string) {
+  if (!mediaRoot) {
+    return key
+  }
+  const mediaRootParts = mediaRoot.split('/').filter((part) => part)
+  if (!mediaRootParts || !mediaRootParts[0]) {
```

Comment:
Haven't tested this yet, but would this return key if media root starts with `/`? ( mediaRootParts[0] would be '' )

Reply:
good catch - pushed a fix